### PR TITLE
Isolated python REPL based on docker containers

### PR DESCRIPTION
**Pull request by piotr-grodek-dsai**

### CODE REVIEW

File: libs/langchain/tests/integration_tests/utilities/test_python_docker_repl.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# NOTE: ports are set to be different in each test so they don't collide
+# as sockets might not be available immediately after container is stopped.
+# It means that tests might also fail to run if port is not freed in time!
+from langchain.utilities.python_docker_repl import PythonContainerREPL
+
+
+def test_python_container_repl_can_be_started() -> None:
+    repl = PythonContainerREPL(port=7120)
+    assert repl is not None
+    del repl
```
Comment (eryk-dsai) [inline libs/langchain/tests/integration_tests/utilities/test_python_docker_repl.py:10]: should `repl` be deleted in other test functions as well ? Probably in `try: ... finally: ...`

**Review by eryk-dsai — APPROVED** [review]

**Comment by maks-operlejn-ds** [comment]
LGTM, but some checks failed

File: libs/langchain/tests/integration_tests/utilities/test_python_docker_repl.py (hunk 1 above)
Comment (rAum) [reply libs/langchain/tests/integration_tests/utilities/test_python_docker_repl.py:10]: I should add a comment, here I wanted to ensure that it does not fail when deleted & gets out of scope.

there is side effect when object is destroyed, the container is closed - adding del is a way to force this behaviour earlier so it is not necessary in other cases.
I also wanted to make tests which showcase usage, adding try..finally would suggest these blocks are necessary but it is untrue.

**Comment by eryk-dsai** [comment]
TODO: after creating the PR to langchain, ask Harrison,  how to best include the docker as a dependency

**Pull request closed without merging**